src/content/blog/2023-10-10-flat-config-rollout-plans.md

```diff
@@ -0,0 +1,75 @@
+teaser: "We have been working on the new configuration system for several years. Here's how we plan to roll it out."
+tags:
+  - Configuration
+  - API
+authors:
+  - nzakas
+categories:
+  - API Changes
+---
+
+It may seem hard to believe, but the [RFC for ESLint's new configuration system](https://github.com/eslint/rfcs/tree/main/designs/2019-config-simplification), nicknamed flat config, was first written in 2019. It took until 2022 ([v8.21.0](https://eslint.org/blog/2022/08/eslint-v8.21.0-released/)) for us to release an experimental, opt-in version of flat config. Since then, we've been making changes and improvements based on feedback from the community. The plan was always to allow the current configuration system, nicknamed eslintrc, to exist alongside flat config for a period of time to allow a smooth transition for ESLint users. With flat config now feature complete, we are happy to share the rollout plan.
+
+## Flat config by default in ESLint v9.0.0
+
+When ESLint v9.0.0 is released, either the end of this year or beginning of next year, flat config will be the default configuration system and we will deprecate, but not remove, eslintrc. New features will be added only for flat config, so we encourage everyone to move off of eslintrc as quickly as possible to take advantage of everything v9.0.0 will offer.
+
+What this change means for you depends on how you use ESLint, and if you have any questions or concerns, please stop by our [Discord](https://eslint.org/chat) to discuss with the team.
+
+**For CLI users**, this means a few things:
+
+* The ESLint CLI will search for `eslint.config.js` by default instead of a `.eslintrc.*` file.
+* If an `eslint.config.js` file is not found, the CLI will consider this an error and won't run.
+* If you want to use eslintrc, you'll need to set the `ESLINT_USE_FLAT_CONFIG` environment variable to `false`. When you do this, you'll get a deprecation warning in the console.
+
+**For rule developers**, the `RuleTester` class will be equivalent to the current `FlatRuleTester` class. If you are passing any parser-related options to your rule tests, you'll need to update those to comply with the flat config format for those options.
+
+You should also make sure that your rules aren't using `context.parserOptions` and `context.parserPath`. Instead, you should be using `context.languageOptions` and `context.languageOptions.parser`, which also work when ESLint is run in eslintrc mode. See our [previous post](https://eslint.org/blog/2023/09/preparing-custom-rules-eslint-v9/) for more information.
+
+**For shareable config authors**, you should update your shareable configs to use flat config format. This includes moving referenced plugins from the `peerDependencies` section of `package.json` to the `dependencies` section.
+
+**For plugin developers**, you should update your exported configs to use flat config format. The rest of your plugin, including rules and processors, don't require any changes.
```

Comment:
I understand that "The rest of your plugin, including rules..." refers to the format of exported objects, but this might be confusing because rule implementations may need some changes as described in "For rule developers".

Perhaps we could add backticks so that this refers to the properties of plugin objects?

```suggestion
**For plugin developers**, you should update your exported `configs` to use flat config format. The rest of your plugin, including `rules` and `processors`, don't require any changes.
```